### ISSUE TITLE
Cucumber Report not getting generate in ADO

### ISSUE THREAD

**Issue by AmrutaSel1** [maciejmaciejewski/azure-pipelines-cucumber#15]
Hello I was trying to get Report
In My framework i am using Cucumber Advanced Report
But in ADO when i add this Extension,"Cucumber" tab is visible but failed to get report. What path i should gave for Input and Output Directory?
Please help me....
Thanks in Advance

![image](https://user-images.githubusercontent.com/65641671/82403239-fcbb2700-9a7b-11ea-99f9-d9b5b3c11ffd.png)

![image](https://user-images.githubusercontent.com/65641671/82403272-10ff2400-9a7c-11ea-8994-3423fe4fc618.png)

**Comment by maciejmaciejewski** [comment]
@AmrutaSel1 first of all you need to save your test results as JSON file using cucumber json reporter and then provide a path to directory where this file is located.
Output directory can be basically anything, task will take generated file and upload it to the storage.
For more information please set build variable system.debug = true and check the logs of the cucumber publish report task.

**Comment by surajrmysuru** [comment]
Hi @maciejmaciejewski ,

I am having a similar issue. My cucumber.json file is available in path .target/cucumber/cucumber.json. 

When I run the job, I am getting the below error
##[warning]Error: Not found outputPath: P:\A5\_w\90\s\$(Sytem.DefaultWorkingDirectory)\target\cucumber

Can you tell me what the problem is here? I have tried couple of things like changing the path of cucumber.json file but it didn't work.


Thanks,
Suraj

**Comment by maciejmaciejewski** [comment]
@surajrmysuru 
Can you please fix typo Sysem -> System and try again?

**Comment by qela** [comment]
Hi @maciejmaciejewski  I am getting the same issue as above, it was originally working and now has stopped for some reason. The cucumber.json file is being created and the output shows the correct logs. 

The cucumber.json file gets created through my code, and then the cucumber.html.json gets created using the Cucumber HTML report task, is that normal? Shouldn't it just be a cucumber.html instead of two files? 

I have attached my log file, hope you can help! 

[PublishCucumberTask.txt](https://github.com/maciejmaciejewski/azure-pipelines-cucumber/files/5083980/PublishCucumberTask.txt)

![image](https://user-images.githubusercontent.com/33481901/90393244-4372ea00-e088-11ea-9aa5-4da6f3f8a50a.png)

**Comment by maciejmaciejewski** [comment]
hey @qela 
Third party library library that I use for generating html report first merges all json files in the directory and then saves it as `cucumber.html.json` so it is fine.

Also your logs looks good so we can assume that the upload task works properly, however for some reason the web component is having issues displaying the content. if you could, please check for any sign of errors in your browser console.

**Comment by qela** [comment]
Hey @maciejmaciejewski apologies just saw your comment. 
That's the thing, even the browser console is fine, the below is seen: 

🚀 Cucumber HTML report C:\test\1\s\target\cucumber.html generated successfully 👍
##[debug]Processed: ##vso[task.addattachment type=cucumber.report;name=cucumber_report.html;]C:\test\1\s\target\cucumber.html
Finishing: Publish Cucumber Report

I'm not really sure what the issue is, overall it looks like it should have uploaded the report but Azure DevOps is having issues displaying it. Is there anything else I can try?

**Comment by maciejmaciejewski** [comment]
@qela
Can you check browser console log when navigating to cucumber tab?
Alternatively, If your report does not contain any sensitive information please zip it and mail it to me so I will try to reproduce it on my azure dev environment.

**Comment by qela** [comment]
@maciejmaciejewski 
Unfortunately I'm not able to send the report, but please see attached for the console log. Is there any chance this could be due to the size of the json file? I noticed my file was around 139MB, so I switched from png to jpg but it still had the same issue. 
[CucumberErrorConsoleLog.txt](https://github.com/maciejmaciejewski/azure-pipelines-cucumber/files/5164146/CucumberErrorConsoleLog.txt)

**Comment by maciejmaciejewski** [comment]
You just reminded me that there is a limit of the attachment size. It is kind of a weird because it does not give any kind of error, attachment just won't load. 
I guess such size is caused by screenshots which by default are kept as embedded base64 encoded strings.

**Comment by qela** [comment]
Yeah it appears it is able to process it and get it onto Azure DevOps, but isn't able to load it. 
Those screenshots are an important part of the report, what is the size limit? Is there any way we are able to get around this?

**Comment by maciejmaciejewski** [comment]
@qela 
I am working on two new features
- ability to use multiple smaller reports on tabs instead of a huge one
- loading screenshots as external resources instead of embedded base64 to decrease report size

Since I am doing it in my free time implementing it might take some time.
What I would suggest is to change reporting behaviour for the time being to take screenshots only in most crucial parts of your web app or take screenshots only at test failures.

**Comment by qela** [comment]
@maciejmaciejewski that's great, thank you for aiming to implement those, that will be a great new piece of functionality! 
Yes I will be either switching to only failure screenshots or splitting up my features based on functionality with the aim that this shall decrease the attachment size.

**Comment by maciejmaciejewski** [comment]
Hello @qela and @AmrutaSel1 
I have just pushed changes with pull request #20 which uses images instead of inline base64 data. Having this change in place allows to load as much screenshots as you want, also the load time and overall plugin performance has been improved.

**Comment by davidgarratt** [comment]
> Hello @qela and @AmrutaSel1
> I have just pushed changes with pull request #20 which uses images instead of inline base64 data. Having this change in place allows to load as much screenshots as you want, also the load time and overall plugin performance has been improved.

Would chunked uploading of the json resolve the issue?

**Comment by maciejmaciejewski** [comment]
@davidgarratt 
The size limit of single file uploaded to azure blob storage is 100MB. With multiple screenshot encoded as inline base64 the size of the file can quickly get out of hand  However there is not such limit when using multiple files so using attachments and chunking size of the json should resolve all loading issues.

**Comment by alspcic** [comment]
@qela @maciejmaciejewski i am getting report generated and published message, however was seeing the same error on "Cucumber" tab. I was using old edge (as a company policy) but when i switched on Edge on Chromium i could see all the reports there. Probably same will work on Chrome

**Comment by naz1719** [comment]
Did this issue fix?
I faced the same issue,
Raised - https://github.com/maciejmaciejewski/azure-pipelines-cucumber/issues/36
